app/models/manageiq/providers/vmware/infra_manager/orchestration_template.rb

```diff
@@ -66,6 +66,16 @@ def workflow_helper
     @workflow_helper ||= MiqProvisionOrchWorkflow.new({:src_vm_id => [id]}, User.current_user, :skip_dialog_load => true, :initial_pass => true)
   end
 
+  def target_name_valid?(name, ems_folder_id = nil)
+    # TODO: need a way to tell the ovf template is for VM template (where target is a VM) or vApp template (where target is a resource pool with VMs)
+    folder = EmsFolder.find_by(:id => ems_folder_id)
+    if folder
+      folder.vms.none? { |vm| vm.name == name }
+    else
+      Vm.find_by(:name => name, :ems_id => ext_management_system.id).blank?
```

Comment:
If we have to access the ems anyway, then `ems.vms.where(:name => name).exists?`

Now that I wrote that it seems the 2 code paths are similar so maybe...

```ruby
parent = EmsFolder.find_by(:id => ems_folder_id) || ext_management_system
parent.vms.where(:name => name).exists?
```